src/azure-cli-testsdk/azure/cli/testsdk/utilities.py

```diff
@@ -157,18 +151,48 @@ def process_response(self, response):
         return response
 
 
-class AADGraphUserReplacer:
+class MSGraphClientPasswordReplacer(RecordingProcessor):
+    """Replace 'secretText' property in 'addPassword' API's response."""
+
+    PWD_REPLACEMENT = 'replaced-microsoft-graph-password'
+
+    def __init__(self):
+        self._activated = False
+
+    def reset(self):
+        self._activated = False
+
+    def process_request(self, request):
+        if request.body and self.PWD_REPLACEMENT in _byte_to_str(request.body):
+            return request
+        if request.path.endswith('/addPassword') and request.method.lower() == 'post':
+            self._activated = True
+        return request
+
+    def process_response(self, response):
+        if self._activated:
+            import json
+
+            body = json.loads(response['body']['string'])
+            body['secretText'] = self.PWD_REPLACEMENT
+
+            response['body']['string'] = json.dumps(body)
+            self._activated = False
+
+        return response
+
+
+class MSGraphUserReplacer:
```

Comment:
Could you please tell why `MSGraphUserReplacer` doesn't need derive from `RecordingProcessor`?

Reply:
```suggestion
class MSGraphUserReplacer(RecordingProcessor):
```